server/tests/src/test/java/org/infinispan/server/extensions/PojoMarshalling.java

```diff
@@ -0,0 +1,36 @@
+package org.infinispan.server.extensions;
+
+import static org.junit.Assert.assertEquals;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.commons.marshall.JavaSerializationMarshaller;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 12.1
+ **/
+public class PojoMarshalling {
+
+   @ClassRule
+   public static final InfinispanServerRule SERVERS = ExtensionsIT.SERVERS;
+
+   @Rule
+   public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+   @Test
+   public void testPojoMarshalling() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.addJavaSerialAllowList(".*");
+      RemoteCache<String, Person> cache = SERVER_TEST.hotrod().withClientConfiguration(builder).withMarshaller(JavaSerializationMarshaller.class).create();
+      cache.put("123", new Person("Enrique", 29));
+      Person person = cache.get("123");
+      assertEquals("Enrique", person.getName());
+      assertEquals(29, person.getAge());
+   }
```

Comment:
Maybe I'm blind, but I can't see where you have configured the cache so that the value is stored as `x-java-object` on the server, e.g.:

```xml
      <distributed-cache name="pojo-personal-shoppers" statistics="true">
         <encoding>
            <key media-type="application/x-java-object"/>
            <value media-type="application/x-java-object"/>
         </encoding>
      </distributed-cache>
```

AFAICT the keys and values will just be stored as a `byte[]` on the server.

Reply:
@ryanemerson done